### DISCUSSION TITLE
xunit integration with MAUI

### DISCUSSION THREAD

**Discussion by ssemanti** [dotnet/maui#2131] [Q&A]
Hi,

We tried to integrate xunit with MAUI project and facing some challenges. Please find the details as following
**Environment:**
Visual Studio 2022 (17.0.0 Preview 3.0)
Dotnet SDK version - 6.0.100-preview.7.21379.14

**Steps followed:**

- Created one MAUI project targeted to .NET 6.0 (preview) framework
- Created one ViewModel inside the project
- Created one Test project (xunit) targeted to .NET 6.0 (preview) framework with one Test class
- Referenced the MAUI project inside the Test project dependencies
- For both the projects the Target OS is Android version 30.0
- Build succeeded and tried to run the test cases from the test explorer window

**Error:**

> Microsoft.VisualStudio.TestPlatform.ObjectModel.TestPlatformException: Testhost process exited with error: It was not possible to find any compatible framework version
The framework 'Microsoft.Android', version '30.0.100-preview.7.110' was not found.
  - No frameworks were found.
You can resolve the problem by installing the specified framework and/or SDK.
The specified framework can be found at:
  - https://aka.ms/dotnet-core-applaunch?framework=Microsoft.Android&framework_version=30.0.100-preview.7.110&arch=x64&rid=win10-x64
. Please check the diagnostic logs for more information.
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyOperationManager.ThrowOnTestHostExited(Boolean testHostExited)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyOperationManager.SetupChannel(IEnumerable`1 sources, String runSettings)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyDiscoveryManager.DiscoverTests(DiscoveryCriteria discoveryCriteria, ITestDiscoveryEventsHandler2 eventHandler)

However if we try to run the Test project as a standalone project (not referencing any MAUI project) and target platform set to none then the Test class is executing successfully.
Please let me know if anyone has faced similar kind of issue and has solution for the same.

Thank you!

**Comment by charleszh-dev** [comment]
Do you want to use xUnit for unit testing or integration testing/device testing? If it is a unit test, then the TargetFramework of the test project may only be set to net6.0. If you are doing Device testing, [XHarness](https://github.com/dotnet/xharness) may be able to help you. You can refer to [maui TestUtils](https://github.com/dotnet/maui/tree/main/src/TestUtils) and [Booster](https://github.com/realZhangChi/Booster) to develop your Device test project.

**Comment by carlosmunozrodriguez** [comment]
I am having the same problem.
For now I was thinking into moving viewModels into its own project that the MAUI project can reference and also reference this new project from the unit test project.
